### PR TITLE
Removes FailedRerouteAllocation and StartedRerouteAllocation

### PR DESCRIPTION
**Pull request by abeyad**
This PR removes the FailedRerouteAllocation and StartedRerouteAllocation
classes, as they were just wrappers for RerouteAllocation that stored
started and failed shards, but these started and failed shards can
be passed in directly to the methods that needed them, removing the
need for this wrapper class and extra level of indirection.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java [hunk 1]
```diff
@@ -61,7 +60,6 @@
     private final GatewayAllocator gatewayAllocator;
     private final ShardsAllocator shardsAllocator;
     private final ClusterInfoService clusterInfoService;
-    private final ClusterName clusterName;
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java]: good catch

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by abeyad** [comment]
Thanks @ywelsch

**Comment by abeyad** [comment]
5.x commit: 23a33eb87db1a165106dfcfeaff779277c8bc159

**Pull request merged**